java/flight/flight-sql/src/test/java/org/apache/arrow/flight/sql/example/FlightSqlExample.java

```diff
@@ -911,14 +912,40 @@ public Runnable acceptPutPreparedStatementQuery(CommandPreparedStatementQuery co
     return () -> {
       assert statementContext != null;
       PreparedStatement preparedStatement = statementContext.getStatement();
+      JdbcParameterBinder binder = null;
 
       try {
         while (flightStream.next()) {
           final VectorSchemaRoot root = flightStream.getRoot();
-          final JdbcParameterBinder binder = JdbcParameterBinder.builder(preparedStatement, root).bindAll().build();
+          binder = JdbcParameterBinder.builder(preparedStatement, root).bindAll().build();
```

Comment:
How are we dealing with old clients that don't yet support stateless prepared statements?